File: apps/mobile/lib/shared/household_tabs.dart
/// Same six household tabs as web Onboard / Account.
const householdTabIds = [
  'owner',
  'family',
  'parking',
  'two_wheeler',
  'four_wheeler',
  'gas',
];

const householdTabLabels = {
  'owner': 'Owner',
  'family': 'Family',
  'parking': 'Parking Details',
  'two_wheeler': 'Two-wheelers',
  'four_wheeler': 'Four-wheelers',
  'gas': 'Gas',
};

String householdSubmitLabel(String tab) {
  switch (tab) {
    case 'family':
      return 'Save household counts';
    case 'parking':
      return 'Save parking';
    case 'gas':
      return 'Save gas';
    case 'two_wheeler':
    case 'four_wheeler':
      return 'Save vehicles';
    default:
      return 'Save profile';
  }
}

File: packages/ui/src/household-tabs.ts
export type HouseholdTabId =
  | "owner"
  | "family"
  | "parking"
  | "two_wheeler"
  | "four_wheeler"
  | "gas";

export const HOUSEHOLD_TABS: Array<{ id: HouseholdTabId; label: string }> = [
  { id: "owner", label: "Owner" },
  { id: "family", label: "Family" },
  { id: "parking", label: "Parking Details" },
  { id: "two_wheeler", label: "Two-wheelers" },
  { id: "four_wheeler", label: "Four-wheelers" },
  { id: "gas", label: "Gas" },
];

export function householdSubmitLabel(
  tab: HouseholdTabId,
  options?: { hasOwner?: boolean; isOwnerTabEdit?: boolean },
): string {
  if (tab === "family") return "Save household counts";
  if (tab === "parking") return "Save parking";
  if (tab === "gas") return "Save gas";
  if (tab === "two_wheeler" || tab === "four_wheeler") return "Save vehicles";
  if (tab === "owner" && options?.isOwnerTabEdit) {
    return options.hasOwner ? "Update owner" : "Save owner";
  }
  return "Save profile";
}
